cardano-cli/src/Cardano/CLI/Read.hs

```diff
@@ -764,7 +774,77 @@ readVoteFile
   -> VoteFile In
   -> IO (Either VoteError (VotingEntry era))
 readVoteFile w fp =
-  first VoteErrorFile <$> conwayEraOnwardsConstraints w (readFileTextEnvelope AsVotingEntry fp)
+  conwayEraOnwardsConstraints w
+    $ first VoteErrorFile <$> readFileTextEnvelope AsVotingEntry fp
+
+emptyVotingProcedures :: VotingProcedures era
+emptyVotingProcedures = VotingProcedures $ Ledger.VotingProcedures Map.empty
+
+singletonVotingProcedures :: ()
+  => ShelleyBasedEra era
+  -> Ledger.Voter (Ledger.EraCrypto (ShelleyLedgerEra era))
+  -> Ledger.GovActionId (Ledger.EraCrypto (ShelleyLedgerEra era))
+  -> Ledger.VotingProcedure (ShelleyLedgerEra era)
+  -> VotingProcedures era
+singletonVotingProcedures _ voter govActionId votingProcedure =
+  VotingProcedures
+    $ Ledger.VotingProcedures
+    $ Map.singleton voter
+    $ Map.singleton govActionId votingProcedure
+
+mergeVotingProcedures :: ()
+  => VotingProcedures era
+  -> VotingProcedures era
+  -> VotingProcedures era
+mergeVotingProcedures vpsa vpsb =
+  VotingProcedures
+    $ Ledger.VotingProcedures
+    $ Map.unionWith (Map.unionWith const)
+        (Ledger.unVotingProcedures (unVotingProcedures vpsa))
+        (Ledger.unVotingProcedures (unVotingProcedures vpsb))
+
+fromVoterRole :: ()
+  => Ledger.EraCrypto (ShelleyLedgerEra era) ~ StandardCrypto
+  => ShelleyBasedEra era
+  -> Ledger.Voter (Ledger.EraCrypto (ShelleyLedgerEra era))
+  -> Voter era
+fromVoterRole _ = \case
+  Ledger.CommitteeVoter cred ->
+    VoterCommittee (VotingCredential (Ledger.coerceKeyRole cred))    -- TODO: Conway era - We shouldn't be using coerceKeyRole.
+  Ledger.DRepVoter cred ->
+    VoterDRep (VotingCredential cred)
+  Ledger.StakePoolVoter kh ->
+    VoterSpo (StakePoolKeyHash kh)
```

Comment:
This can be deleted when `votingProceduresToTxVotes` is deleted.

Reply:
This can be deleted in a follow up PR :+1: